### ISSUE TITLE
This app is crushing on Homeyypro

### ISSUE THREAD

**Issue by treforparr** [AdyRock/com.smartthings#45]
Come up with a crash notification seen all devices have crashed

**Comment by treforparr** [comment]
60bdff3c-96ca-48d5-be4e-133714cc342f

**Comment by treforparr** [comment]
Error code 9210d507-6649-4a12-99d1-967bb0efa39e

**Comment by bashthekeys** [comment]
Same issue, another report here : 0c1adeba-4c8b-454f-9559-9f9af12d0f6a

The error shown on the crash notification is CPU Warning Limit Reached.

**Comment by treforparr** [comment]
What does that mean?

**Comment by AdyRock** [comment]
The logs were showing errors triggering a Flow card as the parameters didn't match. I think that was putting a load on the CPU.
I have now fixed that in the new test version 2.0.21, so could you try that to see if it solves the problem.

**Comment by treforparr** [comment]
It went on and off-line twice againBruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 18:25, Adrian Rockall ***@***.***> wrote:﻿
The logs were showing errors triggering a Flow card as the parameters didn't match. I think that was putting a load on the CPU.
I have now fixed that in the new test version 2.0.21, so could you try that to see if it solves the problem.

**Comment by treforparr** [comment]
26659fe6-0251-4a24-a145-342858391252Bruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 18:38, Trefor Parr ***@***.***> wrote:﻿It went on and off-line twice againBruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 18:25, Adrian Rockall ***@***.***> wrote:﻿
The logs were showing errors triggering a Flow card as the parameters didn't match. I think that was putting a load on the CPU.
I have now fixed that in the new test version 2.0.21, so could you try that to see if it solves the problem.

**Comment by treforparr** [comment]
Do you need a subscription? For homey pro Bruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 18:38, Trefor Parr ***@***.***> wrote:﻿It went on and off-line twice againBruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 18:25, Adrian Rockall ***@***.***> wrote:﻿
The logs were showing errors triggering a Flow card as the parameters didn't match. I think that was putting a load on the CPU.
I have now fixed that in the new test version 2.0.21, so could you try that to see if it solves the problem.

**Comment by AdyRock** [comment]
No, once you have bought a Homey Pro, everything except backups is free.

**Comment by treforparr** [comment]
I wonder what’s causing the crash thenBruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 19:12, Adrian Rockall ***@***.***> wrote:﻿
No, once you have bought a Homey Pro, everything except backups is free.

**Comment by treforparr** [comment]
df2b8bc1-ab6a-4611-8e2e-4047bb0f0973Bruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 19:17, Trefor Parr ***@***.***> wrote:﻿I wonder what’s causing the crash thenBruce Arthur Wayne:Postal Address: 1 Derby Row, Vulcan villageNewton the WillowsWA12 8SBMobile: 07301 533008Home phone: (01925) 509083EmailNew ***@***.*** you can contact me my old ***@***.*** 21 Jul 2024, at 19:12, Adrian Rockall ***@***.***> wrote:﻿
No, once you have bought a Homey Pro, everything except backups is free.

**Comment by treforparr** [comment]
41a87bae-2d0f-462f-a240-9d8d23a1cef0

**Comment by treforparr** [comment]
Still doing it

**Comment by AdyRock** [comment]
You are still running the old version.
Go to the store, locate the app and then add test/ to the end of the url and install version 21.

_App ID

ady.smartthings

App Version

v2.0.19

Homey Version

v11.1.2

Stack Trace

Note: this log has been manually submitted by a user. Log ID: 41a87bae-2d0f-462f-a240-9d8d23a1cef0 User Message: Still crashing_

**Comment by bashthekeys** [comment]
I am still seeing the same issue on the 2.0.21 Experimental. I’ll get a log and details shortly when I’m back at my laptop.On 22 Jul 2024, at 07:35, Adrian Rockall ***@***.***> wrote:﻿
You are still running the old version.
Go to the store, locate the app and then add test/ to the end of the url and install version 21.

**Comment by bashthekeys** [comment]
Diagnostics : fa7c6a72-1447-4a82-b6d0-0f507cd7a193

**Comment by treforparr** [comment]
7460af74-c627-45a3-9299-a0bf0948a25eOn 22 Jul 2024, at 09:37, bashthekeys ***@***.***> wrote:﻿
Diagnostics : fa7c6a72-1447-4a82-b6d0-0f507cd7a193

**Comment by AdyRock** [comment]
I have published a new test version. Can you try that to see if it fixes the errors?

**Comment by treforparr** [comment]
Sorry to be a pain still doing itI’ve sent another error logOn 22 Jul 2024, at 17:58, Adrian Rockall ***@***.***> wrote:﻿
I have published a new test version. Can you try that to see if it fixes the errors?

**Comment by bashthekeys** [comment]
V2.0.22 is working, the logs no longer show any errors when polling. I’ll monitor it for a bit and make sure it’s stable, but looking good so far.

**Comment by AdyRock** [comment]
@treforparr It looks like you sent the device log instead of the information log. Could you send it again please.

**Comment by treforparr** [comment]
3e0136f6-24a9-4e63-af50-35237043772eOn 22 Jul 2024, at 18:27, Adrian Rockall ***@***.***> wrote:﻿
@treforparr It looks like you sent the device log instead of the information log. Could you send it again please.

**Comment by treforparr** [comment]
Could it be something to do with my devices in SmartThingsOn 22 Jul 2024, at 18:31, Trefor Parr ***@***.***> wrote:﻿3e0136f6-24a9-4e63-af50-35237043772eOn 22 Jul 2024, at 18:27, Adrian Rockall ***@***.***> wrote:﻿
@treforparr It looks like you sent the device log instead of the information log. Could you send it again please.

**Comment by bashthekeys** [comment]
@AdyRock I can confirm that for me v2.0.22 fixes the crash, and is stable when polling.

**Comment by treforparr** [comment]
3cfeb918-2e1a-4a56-81ec-28cc07e730f3Still not working properly, it’s still crashing sorry to be a painOn 22 Jul 2024, at 19:09, bashthekeys ***@***.***> wrote:﻿
@AdyRock I can confirm that for me v2.0.22 fixes the crash, and is stable when polling.

**Comment by AdyRock** [comment]
@treforparr I'm not seeing any errors in the log, apart from the cpu usage. 
You do have a lot of devices, so maybe it's just stressing Homey too much. 

Could you disable the logs as they require a lot of extra processing and see if it still crashes. Also, try reducing the polling interval.

**Comment by treforparr** [comment]
Could you tell me how to do that cause I have no clue on how to do any of that, I will have a little player with it tomorrow. I’m not feeling 100% today so my mind is not with it. On 22 Jul 2024, at 19:20, Adrian Rockall ***@***.***> wrote:﻿
@treforparr I'm not seeing any errors in the log, apart from the cpu usage.
You do have a lot of devices, so maybe it's just stressing Homey too much.
Could you disable the logs as they require a lot of extra processing and see if it still crashes. Also, try reducing the polling interval.

**Comment by AdyRock** [comment]
OK, the last log showed and error with a good clue so hopefully the new version in the store will fix it.

To change the polling interval and switch off the log, open the app settings page and the options are in there.

**Comment by treforparr** [comment]
Thank you for all your help 

New  email ***@***.***
Or you can contact me my old email 
***@***.***
